src/components/CountryDetails.js

```diff
@@ -0,0 +1,38 @@
+import { useParams } from "react-router-dom"
+import { useEffect, useState } from "react"
+import axios from "axios";
+
+
+function CountryDetails (props) {
+    const { iataCode } = useParams();
+    const [countryDetails, setCountryDetails] = useState({})
+    // console.log('------', countryDetails)
+    useEffect(() => {
+        axios.get(`https://ih-countries-api.herokuapp.com/countries/${iataCode}`)
+            .then( response => {
+                setCountryDetails(response.data) 
+                // console.log('--- get reponse', response.data)
+            })
+            .catch(err => console.log('Error getting data from API', err))
+    }, [])
+
+//   const findCountry = props.filter(code => code.alpha3Code === iataCode)
+//   console.log(findCountry)
+
+
+    return (
+        <div>
+            <h1>Country Details</h1>
+            {/* <h2>{props.countryArr.name.common}</h2> */}
+            {/* <h2>{iataCode.name.common}</h2>
+            <p>Capital:</p>
+            <p>Area:</p>
+            <p>Borders:</p>
+            <h>{props.data.name.common}</h> */}
+        </div>
+            )
```

Comment:
Since you have created a state here that hols one country details (you get that from API) , that is the variable you want to use in jsx, not the countryArray from props -> we are only looking for one country now.

It is important to remember that in react we have to be aware of the timing of each function, and we know that it can take a moment to get data from API, so we have to wait like this:
```
{countryDetails ? <p>{countryDetails.name.official}</p> : <p>Loading Details</p>}
```

OR this
```
{countryDetails && <p>{countryDetails.name.official}</p>}
```


To check if countryDetails exists there is several ways:
- set state to an empty string , then it is easy as empty string is falsy
```
 const [countryDetails, setCountryDetails] = useState('')
```
- set state to an empty array , then you must check array's length
```
 const [countryDetails, setCountryDetails] = useState([])
```

In react we are not attached to the data type of initial state -> we can replace it with whatever we want